test/distributed/test_c10d_common.py

```diff
@@ -1479,6 +1479,15 @@ def comm_fn(tensor, group=None):
 
         self._test_work_wait(tensor, comm_fn=comm_fn)
 
+    def _test_consecutive_comm_work_wait(self, tensor):
+        def comm_fn(tensor, group=None):
+            work1 = dist.all_reduce(tensor, group=group, async_op=True)
+            work1.wait()
```

Comment:
shall we also have a test case to make this `work1.wait()` be waited after work2 produced?

Reply:
I can add that, but I am not sure if we should. Waiting after work2 is produced is only correct because PG internally uses the same stream for communication. If that subtle assumption no longer holds, we could run into race condition, e.g., the 2nd allreduce start reading the memory space before the 1st finishes writing. 

Curious, is there a use case that we need to call `work1.wait()` after `work2` is produced?